packages/hoppscotch-sh-admin/src/components/users/SharedRequests.vue

```diff
@@ -0,0 +1,191 @@
+<template>
+  <div>
+    <div v-if="fetching" class="flex justify-center">
+      <HoppSmartSpinner />
+    </div>
+
+    <div v-else-if="error">{{ t('shared_requests.load_list_error') }}</div>
+
+    <div v-else-if="sharedRequests.length === 0" class="ml-3 mt-5 text-lg">
+      {{ t('users.no_shared_requests') }}
+    </div>
+
+    <div v-else class="mt-10">
+      <HoppSmartTable :list="sharedRequests">
+        <template #head>
+          <tr
+            class="text-secondary border-b border-dividerDark text-sm text-left bg-primaryLight"
+          >
+            <th class="px-6 py-2">{{ t('shared_requests.id') }}</th>
+            <th class="px-6 py-2 w-30">{{ t('shared_requests.url') }}</th>
+            <th class="px-6 py-2">{{ t('shared_requests.created_on') }}</th>
+            <!-- Empty Heading for the Action Button -->
+            <th class="px-6 py-2 text-center">Actions</th>
+          </tr>
+        </template>
+        <template #body="{ list: sharedRequests }">
+          <tr
+            v-for="request in sharedRequests"
+            :key="request.id"
+            class="text-secondaryDark hover:bg-divider hover:cursor-pointer rounded-xl"
+          >
+            <td class="flex py-4 px-7 max-w-50">
+              <span class="truncate">
+                {{ request.id }}
+              </span>
+            </td>
+
+            <td class="py-4 px-7 w-96">
+              {{ sharedRequestURL(request.request) }}
+            </td>
+
+            <td class="py-2 px-7">
+              {{ getCreatedDate(request.createdOn) }}
+              <div class="text-gray-400 text-tiny">
+                {{ getCreatedTime(request.createdOn) }}
+              </div>
+            </td>
+
+            <td class="flex justify-center">
+              <HoppButtonSecondary
+                v-tippy="{ theme: 'tooltip' }"
+                :title="t('shared_requests.open_request')"
+                :to="`${shortcodeBaseURL}/r/${request.id}`"
+                :blank="true"
+                :icon="IconExternalLink"
+                class="px-3 text-emerald-500 hover:text-accent"
+              />
+
+              <UiAutoResetIcon
+                :title="t('shared_requests.copy')"
+                :icon="{ default: IconCopy, temporary: IconCheck }"
+                @icon-clicked="copySharedRequest(request.id)"
+              />
```

Comment:
This component is similar to the implementation in the app. Am I right? Can we use the `HoppSmartItem` component here instead?

https://github.com/hoppscotch/hoppscotch/blob/23afc201a12e221c01f79d731513fcfeff928fdd/packages/hoppscotch-common/src/components/http/Request.vue#L180-L190

Reply:
@jamesgeorge007 I tried this implementation, but I constantly face the issue where clicking on the copy icon triggers all the icons in each row, however the one selected gets copied. Separating it into a new component fixed the problem where the icon from only one row gets triggered. I have also added better functionality to this component where it can be customized and reused. I am open to ways where using SmartItem actually accomplishes the intended functionality here.

Reply:
Gotcha, there will be a shared state problem. Let's go ahead with the component-based approach 👍
It'll be handy when the pattern is used in another part of the app down the lane.